src/Illuminate/Contracts/Filesystem/Filesystem.php

```diff
@@ -30,9 +30,7 @@ public function exists($path);
      * Get the contents of a file.
      *
      * @param  string  $path
-     * @return string
-     *
-     * @throws \Illuminate\Contracts\Filesystem\FileNotFoundException
+     * @return string|null
```

Comment:
Why the change in interface? Could we not continue to throw the exception, so upgrading is easier for everyone?

Reply:
I've explained this in the main comment of the PR.

Reply:
You didn't explain why though, just that it happened? Why do we need this breaking change?